wdl/qc_imputation.wdl

```diff
@@ -122,53 +156,423 @@ workflow qc_imputation {
     }
 }
 
-task paste {
+task vcf_to_bed {
+
+    File vcf
+    String base = sub(sub(basename(vcf, ".vcf"), "_original", ""), ".calls", "")
+    File exclude_samples
+    String include_regex
+    File ref_fasta
+
+    String docker
+
+    command <<<
+
+        set -euxo pipefail
+
+        catcmd="cat"
+        if [[ ${vcf} == *.gz ]] || [[ ${vcf} == *.bgz ]]; then catcmd="zcat"; fi
+        mem=$((`free -m | grep -oP '\d+' | head -n 1`-500))
+
+        # filter by given samples and prefix, align to reference, unpack multiallelics, remove Y/MT/extra contigs, rename variant id to chr1_1234_A_T
+        echo -e "`date`\talign"
+        comm -23 <($catcmd ${vcf} | grep -E "^#CHROM" | head -1 | tr '\t' '\n' | tail -n+10 | grep -E "${include_regex}" | sort) <(cut -f1 ${exclude_samples} | sort) > include_samples.txt
+        bcftools view -m 2 -M 2 -S include_samples.txt ${vcf} -Ov | \
+        awk 'BEGIN{OFS="\t"} $1 ~ "^#"{ print $0;} \
+                    $1 !~ "^#" \
+                    {  \
+                        ## chr is needed in 38 fasta norm next. annoying uncompress/compress but cant avoid. \
+                        sub("chr", "", $1); $1="chr"$1;
+                        ## rid Y/MT/extra contigs
```

Comment:
for chip only analysis, we'd like to keep the Y and MT. 
Can be left out of this scope but ultimately we'd like to have analysis ready chip data out of this pipeline

Reply:
Going to read further but one thought would be that the MT and Y would not be removed at this step but just in time if they are gonna interfere with some step.

Reply:
Y and MT now come through QC with other metrics except HWE applied. Need to still check with a full data run that joint QC Y chr missingness works (PLINK doesn't include females in Y stats but qctool prob does)

Reply:
aren't MT and Y removed as they don't match in the awk in line 187?